platform-tests/src/test/java/org/junit/platform/commons/util/ReflectionUtilsTests.java

```diff
@@ -651,6 +664,59 @@ void findMethodsWithShadowingUsingHierarchyDownMode() throws Exception {
 			MethodShadowingChild.class.getMethod("method5", Long.class));
 	}
 
+	@Test
+	void handleNoneBridgeMethods() throws Exception {
+		assertFalse(Modifier.isPublic(MethodNoBridgeChild.class.getSuperclass().getModifiers()));
+		assertFalse(Modifier.isPublic(MethodNoBridgeChild.class.getModifiers()));
+		Method beforeMethod = MethodNoBridgeChild.class.getMethod("beforeEach");
+		assertFalse(beforeMethod.isBridge());
+		assumeTrue(beforeMethod.isAnnotationPresent(BeforeEach.class));
+		Method afterMethod = MethodNoBridgeChild.class.getMethod("afterEach");
+		assertFalse(afterMethod.isBridge());
+		assumeTrue(afterMethod.isAnnotationPresent(AfterEach.class));
+
+		findMethodsAndExecuteClass(MethodNoBridgeChild.class);
+	}
+
+	@Test
+	void handleBridgeMethods() throws Exception {
+		assertFalse(Modifier.isPublic(MethodBridgeChild.class.getSuperclass().getModifiers()));
+		assertTrue(Modifier.isPublic(MethodBridgeChild.class.getModifiers()));
+		Method beforeMethod = MethodBridgeChild.class.getMethod("beforeEach");
+		assumeTrue(beforeMethod.isBridge());
+		assumeTrue(beforeMethod.isAnnotationPresent(BeforeEach.class));
+		Method afterMethod = MethodBridgeChild.class.getMethod("afterEach");
+		assumeTrue(afterMethod.isBridge());
+		assumeTrue(afterMethod.isAnnotationPresent(AfterEach.class));
+
+		findMethodsAndExecuteClass(MethodBridgeChild.class);
+	}
+
+	private void findMethodsAndExecuteClass(Class<?> testClass) {
+		List<Method> methods = ReflectionUtils.findMethods(testClass, method -> true);
+		assertThat(methods).hasSize(7);
+		Set<String> bridges = methods.stream().filter(Method::isBridge).map(Method::getName).collect(toSet());
+		assertThat(bridges).isEmpty();
+
+		bridgeMethodSequence.clear();
+		// @formatter:off
+		LauncherDiscoveryRequest request = LauncherDiscoveryRequestBuilder
+			.request()
+			.selectors(DiscoverySelectors.selectClass(testClass))
+			.build();
+		// @formatter:on
+		Launcher launcher = LauncherFactory.create();
+		launcher.execute(request);
+		assertAll("bridge method sequence test",
+			() -> assertEquals("static parent.beforeAll()", bridgeMethodSequence.get(0)),
+			() -> assertEquals("parent.beforeEach()", bridgeMethodSequence.get(1)),
+			() -> assertEquals("child.anotherBeforeEach()", bridgeMethodSequence.get(2)),
+			() -> assertEquals("child.test()", bridgeMethodSequence.get(3)),
+			() -> assertEquals("child.anotherAfterEach()", bridgeMethodSequence.get(4)),
+			() -> assertEquals("parent.afterEach()", bridgeMethodSequence.get(5)),
+			() -> assertEquals("static parent.afterAll()", bridgeMethodSequence.get(6)));
+	}
+
```

Comment:
IMHO these tests are fine but in the wrong place. They are integration tests for execution aspects of the Jupiter engine, not unit tests for ReflectionUtils. I think we should change them to directly call ReflectionUtils and move these tests over to junit-jupiter-engine.